account_move_multi_company/models/account_move.py

```diff
@@ -0,0 +1,119 @@
+                    else:
+                        dedicated_companies_vals[
+                            line.transfer_to_company_id]['line_ids'].append(
+                            (0, 0, {'account_id': line.transfer_to_company_id.
+                                    due_to_account_id.id, 'partner_id':
+                                    line.company_id.partner_id.id,
+                                    'credit': line.debit}))
+                        dedicated_companies_vals[
+                            line.transfer_to_company_id]['line_ids'].append(
+                            (0, 0, {'account_id':
+                                    self.env['account.account'].sudo().
+                                    search(
+                                            [('code',
+                                              '=', line.account_id.code),
+                                             ('company_id', '=',
+                                              line.company_id.id)],
+                                        limit=1).id,
+                                    'partner_id': line.partner_id.id,
+                                    'debit': line.debit}))
+                    lines_to_reconcile.append(line)
+
+            # Journal entry in main company for employees who worked for other
+            # companies of the group.
+            if self.env.user.company_id.due_fromto_payment_journal_id \
+                    and transfer_lines:
+                journal_entry_transfer = self.env['account.move'].create({
+                    'journal_id':
+                    self.env.user.company_id.
+                    due_fromto_payment_journal_id.id,
+                    'line_ids': transfer_lines
+                })
+                self += journal_entry_transfer
+
+                transfer_lines = journal_entry_transfer.line_ids.\
+                    filtered("credit").sorted(key=lambda r: r.id)
+
+                for (line, rec_line) in zip(transfer_lines,
+                                            lines_to_reconcile):
+                    (line + rec_line).reconcile()
+
+                # Journal Entry for each dedicated company
+                dedicated_company_move = self.env['account.move'].sudo()
+                for transfer_to_company_id in dedicated_companies_vals:
+                    dedicated_company_move += self.env['account.move'].sudo().\
+                        create(dedicated_companies_vals[
+                               transfer_to_company_id])
+                self += dedicated_company_move
+
+        return super(AccountMove, self).post()
+
+    def _post_validate(self):
+        # Override to prevent ValidationError
+        # Method in odoo/addons/account/models/account_move line 357
+        return True
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    transfer_to_company_id = fields.Many2one(
+        'res.company', string='Transfer to')
```

Comment:
@Khalid-SerpentCS Can you create one file per model?

Ref:
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#file-naming